scripts/estimation.ts

```diff
@@ -0,0 +1,208 @@
+      }
+    )
+
+  _handleGasEstimates(gasEstimateComponents, 'UpgradeExecutor')
+}
+
+async function estimateAll(signer: any) {
+  const nodeInterface = NodeInterface__factory.connect(
+    NODE_INTERFACE_ADDRESS,
+    signer
+  )
+
+  await estimateContractDeployment('Bridge', nodeInterface, [])
+  await estimateContractDeployment('SequencerInbox', nodeInterface, [
+    maxDataSize,
+  ])
+  await estimateContractDeployment('Inbox', nodeInterface, [maxDataSize])
+  await estimateContractDeployment('RollupEventInbox', nodeInterface, [])
+  await estimateContractDeployment('Outbox', nodeInterface, [])
+
+  await estimateContractDeployment('ERC20Bridge', nodeInterface, [])
+  await estimateContractDeployment('ERC20Inbox', nodeInterface, [maxDataSize])
+  await estimateContractDeployment('ERC20RollupEventInbox', nodeInterface, [])
+  await estimateContractDeployment('ERC20Outbox', nodeInterface, [])
+
+  await estimateContractDeployment('BridgeCreator', nodeInterface, [
+    [
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+    ],
+    [
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+      ethers.Wallet.createRandom().address,
+    ],
+  ])
+  await estimateContractDeployment('OneStepProver0', nodeInterface)
+  await estimateContractDeployment('OneStepProverMemory', nodeInterface)
+  await estimateContractDeployment('OneStepProverMath', nodeInterface)
+  await estimateContractDeployment('OneStepProverHostIo', nodeInterface)
+  await estimateContractDeployment('OneStepProofEntry', nodeInterface, [
+    ethers.Wallet.createRandom().address,
+    ethers.Wallet.createRandom().address,
+    ethers.Wallet.createRandom().address,
+    ethers.Wallet.createRandom().address,
+  ])
+  await estimateContractDeployment('ChallengeManager', nodeInterface)
+  await estimateContractDeployment('RollupAdminLogic', nodeInterface)
+  await estimateContractDeployment('RollupUserLogic', nodeInterface)
+  await estimateUpgradeExecutorDeployment(nodeInterface)
+  await estimateContractDeployment('ValidatorUtils', nodeInterface)
+  await estimateContractDeployment('ValidatorWalletCreator', nodeInterface)
+  await estimateContractDeployment('RollupCreator', nodeInterface)
+  await estimateContractDeployment('DeployHelper', nodeInterface)
+}
+
+function _handleGasEstimates(
+  gasEstimateComponents: [BigNumber, BigNumber, BigNumber, BigNumber] & {
+    gasEstimate: BigNumber
+    gasEstimateForL1: BigNumber
+    baseFee: BigNumber
+    l1BaseFeeEstimate: BigNumber
+  },
+  contractName: string
+) {
+  totalGas = totalGas.add(gasEstimateComponents.gasEstimate)
+  totalL1Gas = totalL1Gas.add(gasEstimateComponents.gasEstimateForL1)
+  totalL2Gas = totalL2Gas.add(
```

Comment:
instead of having these global vars, return the calculated scaled gas value and sum them within estimateAll